src/popup/router/pages/FungibleTokens.vue

```diff
@@ -0,0 +1,221 @@
+    },
+    methods:{
+        switchTabs(tab) {
+            this.activeTab = tab
+        },
+        tabAcitveClass(tab) {
+            return this.activeTab == tab ? 'tab-active' : ''
+        },
+        navigateAccount() {
+            this.$router.push('/account')
+        },
+        validate(type) {
+            if(type == 'contract') {
+                this.token.precisionDisabled = false
+                if(this.token.contract.length == 53) {
+                    this.searchTokenMetaInfo(this.token.contract)
+                }
+            }
+        },
+        async next() {
+            let added = this.tokens.find(tkn => tkn.contract == this.token.contract && tkn.parent == this.account.publicKey)
+            if( this.token.contract.length != 53 || 
+                (this.token.symbol.length < 1 || this.token.symbol.length > 12) || 
+                isNaN(this.token.precision) ||
+                (!isNaN(this.token.precision) && (this.token.precision < 0 || this.token.precision > 36 ))
+            ) {
+                this.$store.dispatch('popupAlert', { name: 'account', type: 'token_add'})
+            }else if(typeof added != 'undefined'){
+                this.$store.dispatch('popupAlert', { name: 'account', type: 'token_exists'})
+            }else {
+                this.loading = true
+                let call = await this.sdk.contractCall(FUNGIBLE_TOKEN_CONTRACT,this.token.contract,'balance',[this.account.publicKey])
+                let balance = await call.decode()
+                this.loading = false
+                this.token.balance = balance == 'None' ? 0 : balance.Some[0]
+                this.addStep = true
+            }
+        },
+        async addCustomToken() {
+            let tokens = this.tokens.map(tkn => tkn)
+            tokens.push({
+                contract:this.token.contract,
+                name:this.token.name,
+                symbol:this.token.symbol,
+                precision:this.token.precision,
+                balance:this.token.balance,
+                parent:this.account.publicKey
+            })
+            this.$store.dispatch('setTokens', tokens).then(() => {
+                browser.storage.sync.set({ tokens: this.tokens}).then(() => { 
+                    this.$store.dispatch('popupAlert', {
+                            name: 'account',
+                            type: 'added_success'
+                        }).then(() => { 
+                            this.token.contract = ""
+                            this.token.symbol = ""
+                            this.token.precision = 0
+                            this.token.precisionDisabled = false
+                            this.addStep = false
+                        });
+                });
+            });
+        },
+        searchTokenMetaInfo(address) {
+            this.loading = true
+            try {
+                this.sdk.contractCall(FUNGIBLE_TOKEN_CONTRACT,address,'meta_info')
```

Comment:
@mradkov Can you check this calling of contract method meta_info. Although the function is public and do not change the state still takes AE for transaction. I tried setting gas and gasPrice to 0 but then the method cannot be called at all.

Reply:
I’ll review it